memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java

```diff
@@ -119,21 +117,29 @@ public void initConversationMetaIndexIfAbsent(ActionListener<Boolean> listener)
     /**
      * Adds a new conversation with the specified name to the index
      * @param name user-specified name of the conversation to be added
+     * @param applicationType the application type that creates this conversation
      * @param listener listener to wait for this to finish
      */
-    public void createConversation(String name, ActionListener<String> listener) {
+    public void createConversation(String name, String applicationType, ActionListener<String> listener) {
         initConversationMetaIndexIfAbsent(ActionListener.wrap(indexExists -> {
             if (indexExists) {
-                String userstr = userstr();
+                String userstr = client
```

Comment:
Why replace all `userstr()` with a longer coding ?

`userstr` seems not a good method name, change to `getUserStrFromThreadContext` ?

Reply:
Updated.